src/lib/updates/updateFeedEvents.js

```diff
@@ -1,50 +1,15 @@
-import { get } from 'lodash'
-import userStorage from '../gundb/UserStorage'
+import wallet from '../wallet/GoodWallet'
 
-const fromDate = new Date('2021/06/17')
+const fromDate = new Date('2021/06/21')
 
 /**
  * fix broken feed items
  * @returns {Promise<void>}
  */
 const updateFeedEvents = async (lastUpdate, prevVersion, log) => {
-  await userStorage.ready
-  await userStorage.wallet.ready
-  const feed = await userStorage.getAllFeed()
-  log.info(`checking ${feed.length} feed events`)
-  const res = await Promise.all(
-    feed.map(async feedEvent => {
-      const id = feedEvent.id
-
-      const type = feedEvent.type
-      const counterPartyFullName = get(feedEvent, 'data.counterPartyFullName', null)
-      const counterPartyProfile = get(feedEvent, 'data.counterPartyProfile', null)
-      const counterPartyAddress = get(feedEvent, 'data.counterPartyAddress', null)
-      if (
-        id.startsWith('0x') &&
-        type !== 'claim' &&
-        type !== 'bonus' &&
-        (!counterPartyFullName || !counterPartyProfile || !counterPartyAddress)
-      ) {
-        const receipt = await userStorage.wallet.getReceiptWithLogs(id).catch(e => {
-          log.warn('no receipt found for id:', e.message, e, id)
-          return undefined
-        })
-
-        if (receipt) {
-          log.info('missing data in feedevent, processing receipt again', { id, feedEvent, receipt })
-          const event = await userStorage.feedStorage.handleReceipt(receipt)
-          log.info('updated feedEvent:', { id, event })
-          return event
-        }
-        log.warn('getFeedPage no receipt found for feedEvent id:', id)
-        return null
-      }
-    }),
-  )
-
-  const filtered = res.filter(x => !!x)
-  log.info(`updated ${filtered.length} feed events`, filtered)
-  return filtered
+  log.info('waiting for wallet init')
+  await wallet.ready
+  log.info('wallet ready, syncing blockchain')
+  return wallet.syncTxWithBlockchain(6000000)
```

Comment:
is there a way to find the block number by date ? 
if yes - let's find the block corresponding to the fromDate defined here and will use its number instead of the fixed value

Reply:
not possible. you can only guesstimate (ie fuse has roughly 1 block per 5 seconds)
anyways we know the exact  block number  which we first  released  the protocol on which is around  6.2M.  6M is fine

Reply:
Okay. I supposed it may be possible because the blockchain is a linked list structure having next/prev references and block date . So, having some block we maybe could get the previous one, check its date and repeat until reach the date we need

But if no - then okay, approving  the current solution